content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -86,46 +86,325 @@ In the above example, this version is `kubelet.config.k8s.io/v1beta1`.
 
 ## Drop-in directory for kubelet configuration files {#kubelet-conf-d}
 
-As of Kubernetes v1.28.0, the kubelet has been extended to support a drop-in configuration directory. The location of it can be specified with
-`--config-dir` flag, and it defaults to `""`, or disabled, by default.
+{{<feature-state for_k8s_version="v1.30" state="beta" >}}
 
-You can only set `--config-dir` if you set the environment variable `KUBELET_CONFIG_DROPIN_DIR_ALPHA` for the kubelet process (the value of that variable does not matter).
-For Kubernetes v{{< skew currentVersion >}}, the kubelet returns an error if you specify `--config-dir` without that variable set, and startup fails.
-You cannot specify the drop-in configuration directory using the kubelet configuration file; only the CLI argument `--config-dir` can set it.
+You can specify a drop-in configuration directory for the kubelet. By default, the kubelet does not look
+for drop-in configuration files anywhere - you must specify a path.
+For example: `--config-dir=/etc/kubernetes/kubelet.conf.d`
+
+For Kubernetes v1.28 to v1.29, you can only specify `--config-dir` if you also set
+the environment variable `KUBELET_CONFIG_DROPIN_DIR_ALPHA` for the kubelet process (the value
+of that variable does not matter).
 
-One can use the kubelet configuration directory in a similar way to the kubelet config file.
 {{< note >}}
 The suffix of a valid kubelet drop-in configuration file must be `.conf`. For instance: `99-kubelet-address.conf`
 {{< /note >}}
 
-For instance, you may want a baseline kubelet configuration for all nodes, but you may want to customize the `address` field. This can be done as follows:
+The kubelet processes files in its config drop-in directory by sorting the **entire file name** alphanumerically.
+For instance, 00-kubelet.conf is processed first, and then overridden with a file named 01-kubelet.conf.
+
+These files may contain partial configurations and might not be valid config files by themselves.
+Validation is only performed on the final resulting configuration file.
+This offers you flexibility in how you manage and combine kubelet configuration that comes from different sources.
+However, it's important to note that the behavior varies based on the data type of the configuration fields.
+Here are some examples of how different data types behave during configuration merging:
```

Comment:
The details around configuration merging are specific and aren't right for a task page. Please move them into https://kubernetes.io/docs/reference/node/ so that people wanting to configure the kubelet can still understand the overall task.